admin/desk/utils.py

```diff
@@ -0,0 +1,185 @@
+"""
+Modified from https://gist.github.com/pamelafox/5855372
+"""
+
+import requests
+import json
+from requests_oauthlib import OAuth1Session
+
+from django.conf import settings
+
+
+class DeskError(Exception):
+    pass
+
+
+class DeskClient(object):
+    """ Initialize the client with the given sitename, username,
+    and password. The suggested way to do this in the Desk docs for
+    a single person is to use Basic Auth.
+    """
+    BASE_URL = 'desk.com/api/v2'
+    SITE_NAME = 'openscience'
+
+    def __init__(self):
+        self.oauth = OAuth1Session(
+            settings.DESK_KEY,
+            client_secret=settings.DESK_KEY_SECRET,
+            resource_owner_key=settings.DESK_TOKEN,
+            resource_owner_secret=settings.DESK_TOKEN_SECRET
+        )
+
+    def build_url(self, service):
+        """ Constructs the URL for a given service."""
+        return 'https://%s.%s/%s' % (self.SITE_NAME, DeskClient.BASE_URL, service)
+
+    def call_get(self, service, params=None):
+        """ Calls a GET API for the given service name and URL parameters."""
+        url = self.build_url(service)
+        r = self.oauth.get(url, params=params)
+        if r.status_code != requests.codes.ok:
+            raise DeskError(str(r.status_code))
+        return json.loads(r.content)
+
+    def call_post(self, service, data=None):
+        """ Calls a POST API for the given service name and POST data."""
+        url = self.build_url(service)
+        r = self.oauth.post(url, data=json.dumps(data))
+        if r.status_code >= 400:
+            raise DeskError(str(r.status_code))
```

Comment:
I can certainly do that, but in my experience Desk API really doesn't tell you anything in the body.